include/deal.II/base/symmetric_tensor.h

```diff
@@ -2320,6 +2324,401 @@ Number second_invariant (const SymmetricTensor<2,3,Number> &t)
+ * tensor $T$.
+ *
+ * The specialized algorithms utilized in computing the eigenvectors are
+ * presented in
+ * @code{.bib}
+ * @Article{Kopp2008,
+ *   title        = {Efficient numerical diagonalization of hermitian 3x3 matrices},
+ *   author       = {Kopp, J.},
+ *   journal      = {International Journal of Modern Physics C},
+ *   year         = {2008},
+ *   volume       = {19},
+ *   number       = {3},
+ *   pages        = {523--548},
+ *   doi          = {10.1142/S0129183108012303},
+ *   eprinttype   = {arXiv},
+ *   eprint       = {physics/0610206v3},
+ *   eprintclass  = {physics.comp-ph},
+ *   url          = {https://www.mpi-hd.mpg.de/personalhomes/globes/3x3/index.html}
+ * }
+ * @endcode
+ *
+ * @relates SymmetricTensor
+ * @author Joachim Kopp, Jean-Paul Pelteret, 2017
+ */
+template <typename Number>
+std::array<std::pair<Number, Tensor<1,2,Number> >,2>
+eigenvectors (const SymmetricTensor<2,2,Number> &T,
+              const enum EigenvectorMethod       method = EigenvectorMethod::ql_implicit_shifts)
+{
+  std::array<std::pair<Number, Tensor<1,2,Number> >,2> eig_vals_vecs;
+
+  if (method == EigenvectorMethod::hybrid)
+    eig_vals_vecs = internal::hybrid(T);
+  else if (method == EigenvectorMethod::ql_implicit_shifts)
+    eig_vals_vecs = internal::ql_implicit_shifts(T);
+  else if (method == EigenvectorMethod::jacobi)
+    eig_vals_vecs = internal::jacobi(T);
+  else
+    AssertThrow(false, ExcNotImplemented());
+
+  std::sort(eig_vals_vecs.begin(), eig_vals_vecs.end(),
+            internal::SortEigenValuesVectors<2,Number>());
+  return eig_vals_vecs;
+}
+
+
+
+/**
+ * Return the eigenvalues and eigenvectors of a real-valued rank-2 symmetric
+ * tensor $T$.
+ *
+ * The specialized algorithms utilized in computing the eigenvectors are
+ * presented in
+ * @code{.bib}
+ * @Article{Kopp2008,
+ *   title        = {Efficient numerical diagonalization of hermitian 3x3 matrices},
+ *   author       = {Kopp, J.},
+ *   journal      = {International Journal of Modern Physics C},
+ *   year         = {2008},
+ *   volume       = {19},
+ *   number       = {3},
+ *   pages        = {523--548},
+ *   doi          = {10.1142/S0129183108012303},
+ *   eprinttype   = {arXiv},
+ *   eprint       = {physics/0610206v3},
+ *   eprintclass  = {physics.comp-ph},
+ *   url          = {https://www.mpi-hd.mpg.de/personalhomes/globes/3x3/index.html}
+ * }
+ * @endcode
+ *
+ * @relates SymmetricTensor
+ * @author Joachim Kopp, Jean-Paul Pelteret, 2017
+ */
+template <typename Number>
+std::array<std::pair<Number, Tensor<1,3,Number> >,3>
+eigenvectors (const SymmetricTensor<2,3,Number> &T,
+              const enum EigenvectorMethod       method = EigenvectorMethod::ql_implicit_shifts)
+{
+  std::array<std::pair<Number, Tensor<1,3,Number> >,3> eig_vals_vecs;
+
+  if (method == EigenvectorMethod::hybrid)
+    eig_vals_vecs = internal::hybrid(T);
+  else if (method == EigenvectorMethod::ql_implicit_shifts)
```

Comment:
not important, but I think if you can enumerate the types, using a `switch` statement may have been easier to read